Doc/reference/compound_stmts.rst

```diff
@@ -1107,7 +1107,7 @@ subject value:
    If only keyword patterns are present, they are processed as follows,
    one by one:
 
-   I. The keyword is looked up as an attribute on the subject.
+   1. The keyword is looked up as an attribute on the subject.
```

Comment:
I expect the different numbering style (Roman numerals) was used here to differentiate this inner numbered list from the outer one.


# Before

<img width="813" alt="image" src="https://github.com/python/cpython/assets/1324225/8d908bc0-76b5-4693-a8e5-73e0c0b46a7b">

# After

<img width="820" alt="image" src="https://github.com/python/cpython/assets/1324225/a6e2e75b-f7c7-45f1-87a4-16f08aa233cf">

Can we keep it them as Roman numerals, but fix the formatting?

Reply:
Sphinx <7 doesn't seem to support them, so they are actually just regular paragraphs starting with `I. ...`.  Letters like `a. b. c. ...` are also not supported, so we only have numbers left and have to rely on the indentation to distinguish the levels.

Reply:
I see.

Well, we need to keep support for older Sphinx for the benefit of Linux distros (and are testing it on CI), but can we use Sphinx 7 for our actual main build and deploy?

The Roman numerals won't cause any errors for for Sphinx <7, and they'll be better rendered for for Sphinx 7.

Re: https://github.com/python/cpython/issues/99380 and cc @AA-Turner.

Reply:
I should be able to fix the indentation/rendering of the nested lists while keeping the roman numerals, the only issue is that since they are rendered as `<p>`s rather than `<li>`s, it's technically incorrect (at least on Sphinx <7).  Not sure if that matters though (maybe for screen readers or similar cases?).

Reply:
Actually if I fix it for <7, once we upgrade to 7 it will break again.

On <7 I could do:
```rst
I. The keyword is looked up as an attribute on the subject.

* ...
* ...
```
which will be seen as a paragraph followed by a list, but on 7 it would have to be:
```rst
I. The keyword is looked up as an attribute on the subject.

   * ...
   * ...
```
since it will be seen as a list item followed by a sublist that needs to be indented.

If I leave the indentation, it will render an additional blockquote around the sublist, and it will cause `sphinx-lint` to complain, so switching to numbers might still be the best compromise.

Reply:
I don't mind too much if the display isn't perfect for 7, as long as it still builds and looks reasonable.

Then we can use 7 for our deploys, and sphinx-lint will be happy too.

Would that work?

Reply:
`sphinx-lint` will be sad on 7 too, since the checker currently ignores alphabetic lists (like Sphinx 6 does) and sees this as an incorrectly indented list under a paragraph, regardless of the Sphinx version used.

I tried adding support for alphabetic lists, but it's a can of worms with many false positives.

Reply:
Hmm, maybe we should ignore the Sphinx Lint warning here?

Reply:
```suggestion
   I. The keyword is looked up as an attribute on the subject.
```

Reply:
I haven't looked into this in a while, but if possible we should find a solution that is both rendered correctly and that is not reported by `sphinx-lint` as error. 
Fixing `sphinx-lint` is also an option if the error is reported mistakenly, but avoiding alphabetic lists and roman numerals might still be a simpler solution.